app/controllers/application_controller/timelines.rb

```diff
@@ -93,9 +93,18 @@ def tl_build_timeline_report_options
       from_dt = create_time_in_utc("#{from_date.strftime} 00:00:00",
                                    session[:user_tz])
 
-      rec_cond, *rec_params = @tl_record.event_where_clause(@tl_options.evt_type)
+      parameters = [from_dt, to_dt]
+      rec_cond = ""
+      rec_params = ""
+
+      if params[:tl_storage_systems].nil? or params[:tl_storage_systems]== [""]
+        rec_cond, *rec_params = @tl_record.event_where_clause(@tl_options.evt_type)
+        parameters = rec_params + [from_dt, to_dt]
+      else
+        rec_cond = tl_map_storage_systems(params[:tl_storage_systems])
```

Comment:
So it looks like if you aren't on a specific record, you want to show all events from all storage systems?
Would it make more sense to have @tl_record be the `Autosde::StorageManager` and keep using the same `@tl_record.event_where_clause` ?

Reply:
I edited the function as you recommended, the function is related to the StorageManager and PhysicalStorage models now.
Changes will be pushed hopefully today.